### PR TITLE
eventing-controller helm chart

### PR DESCRIPTION
**Pull request by k15r**
**Description**

Changes proposed in this pull request:

- run eventing-controller as nonroot
- install eventing-controller

**Related issue(s)**

### CODE REVIEW

File: components/eventing-controller/Dockerfile [hunk 1]
```diff
@@ -20,6 +20,6 @@ LABEL source = git@github.com:kyma-project/kyma.git
 WORKDIR /
 COPY --from=builder /go/src/github.com/kyma-project/kyma/components/eventing-controller/eventing-controller .
 COPY licenses/ /licenses/
-USER nonroot:nonroot
+USER 1001
```
Comment (nachtmaar) [inline components/eventing-controller/Dockerfile:23]: Can you provide some context why this change is required?

File: resources/cluster-essentials/files/subscriptions.eventing.kyma-project.io.crd.yaml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+
```
Comment (nachtmaar) [inline resources/cluster-essentials/files/subscriptions.eventing.kyma-project.io.crd.yaml:1]: Now that the eventing-controller Beb integration is merged, let's use the updated CRD

File: resources/eventing/charts/controller/Chart.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+apiVersion: v2
+name: controller
+description: A Helm chart for Kubernetes
```
Comment (nachtmaar) [inline resources/eventing/charts/controller/Chart.yaml:3]: nit: let's have a real description here :)

File: resources/eventing/charts/controller/Chart.yaml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 0.1.0
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+appVersion: 1.16.0
```
Comment (nachtmaar) [inline resources/eventing/charts/controller/Chart.yaml:23]: I think we are not using appVersion in Kyma yet. Do you plan to use it in this chart in the future? If yes I guess 1.16.0 is not the version of the controller. I think it makes sense to have it if we have proper versioning for the controller.

File: resources/eventing/charts/controller/values.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+# override name to avoid collision with kantive eventing resources
```
Comment (nachtmaar) [inline resources/eventing/charts/controller/values.yaml:1]: kantive -> knative

Comment (nachtmaar) [reply resources/eventing/charts/controller/values.yaml:1]: nit: I am wondering where you expect a collision? Knative eventing-controller is installed into knative-eventing namespace whereas this controller is installed into kyma-system.

File: resources/eventing/charts/controller/values.yaml [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+serviceAccount:
+  # name defines optionally another name than the default name for the service account
+  name: ""
+
+clusterRole:
+  # name defines optionally another name than the default name for the cluster role
+  name: ""
+
+image:
+  # name is the name of the container image for the event-publisher-proxy
```
Comment (nachtmaar) [inline resources/eventing/charts/controller/values.yaml:13]: The event-publisher-proxy is mentioned in the comments. I guess you want to update it to match the controller instead.

Comment (nachtmaar) [inline resources/eventing/charts/controller/values.yaml:13]: The event-publisher-proxy is mentioned in the comments. I guess you want to update it to match the controller instead.

**Review by nachtmaar — COMMENTED** [review]
Review done, however I want to also run it later in the day :)

File: resources/eventing/charts/controller/values.yaml [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+  name: "eventing-controller"
+  # tag is the container tag of the event-publisher-proxy image
+  tag: "PR-9750"
```
Comment (nachtmaar) [inline resources/eventing/charts/controller/values.yaml:16]: Let's use the latest available image from this PR: https://github.com/kyma-project/kyma/pull/9744
```suggestion
  tag: "PR-9744"
```

File: installation/resources/installer-cr-cluster.yaml.tpl [hunk 8]
```diff
@@ -22,8 +22,8 @@ spec:
       namespace: "kyma-installer"
     - name: "istio-kyma-patch"
       namespace: "istio-system"
-    # - name: "eventing"
-    #   namespace: kyma-system
+    - name: "eventing"
```
[the remark this answers is not in the document]
Comment (sayanh) [reply installation/resources/installer-cr-cluster.yaml.tpl]: Reverted back to optional

File: resources/eventing/charts/controller/Chart.yaml (hunk 4 above)
Comment (sayanh) [reply resources/eventing/charts/controller/Chart.yaml:23]: Removed `appVersion` for now

File: components/eventing-controller/Dockerfile (hunk 1 above)
Comment (sayanh) [reply components/eventing-controller/Dockerfile]: Removed the change in Dockerfile

File: resources/cluster-essentials/files/subscriptions.eventing.kyma-project.io.crd.yaml (hunk 2 above)
Comment (sayanh) [reply resources/cluster-essentials/files/subscriptions.eventing.kyma-project.io.crd.yaml]: Done

File: resources/eventing/charts/controller/Chart.yaml (hunk 3 above)
Comment (sayanh) [reply resources/eventing/charts/controller/Chart.yaml:3]: Done

File: resources/eventing/charts/controller/values.yaml (hunk 7 above)
Comment (sayanh) [reply resources/eventing/charts/controller/values.yaml]: Not applicable as we have `PR-9868`

File: resources/eventing/charts/controller/values.yaml (hunk 6 above)
Comment (sayanh) [reply resources/eventing/charts/controller/values.yaml:15]: Changed

File: resources/eventing/README.md [hunk 9]
```diff
@@ -17,8 +22,8 @@ You can install this Helm chart using either Helm or Kyma CLI. In both cases, th
 
 ```bash
 $ cat << EOF > helm-values.yaml
-event-publisher-proxy:
-  upstreamAuthentication:
+eventing:
```
Comment (nachtmaar) [inline resources/eventing/README.md:25]: Domain name is missing in this example:

```suggestion
global:
  domainName: "$domainName"
eventing:
```

File: resources/eventing/README.md [hunk 10]
```diff
@@ -4,11 +4,16 @@ This Helm chart contains all components required for eventing in Kyma:
...
 
 ## Event Publisher Proxy
 
 This component receives Cloud Event publishing requests from the cluster workloads (microservice or Serverless functions) and redirects them to the Enterprise Messaging Service Cloud Event Gateway. Click [here](https://github.com/kyma-project/kyma/tree/master/components/event-publisher-proxy) for more details.
 
+## Controller
+
+This component manages the internal infrastructure in order to receive an event for all subscriptions
+
 ## Install
```
Comment (nachtmaar) [inline resources/eventing/README.md:18]: Let's also add a short line how to install the CRD then installed via helm only:

```
kubectl apply -f resources/cluster-essentials/files/subscriptions.eventing.kyma-project.io.crd.yaml
```

File: resources/eventing/README.md [hunk 11]
```diff
@@ -17,8 +22,8 @@ You can install this Helm chart using either Helm or Kyma CLI. In both cases, th
 
 ```bash
 $ cat << EOF > helm-values.yaml
-event-publisher-proxy:
-  upstreamAuthentication:
+eventing:
+  authentication:
     oauthClientId: "$bebOauthClientId"
     oauthClientSecret: "$bebOauthClientSecret"
     oauthTokenEndpoint: "$bebOauthTokenEndpoint"
```
Comment (nachtmaar) [inline resources/eventing/README.md:29]: - top key should be authentication, not eventing.
- domain name is missing

```suggestion
global:
  domainName: "$domainName"
authentication:
  oauthClientId: "$bebOauthClientId"
  oauthClientSecret: "$bebOauthClientSecret"
  oauthTokenEndpoint: "$bebOauthTokenEndpoint"
```

File: resources/eventing/README.md (hunk 10 above)
Comment (sayanh) [reply resources/eventing/README.md:18]: Done

File: resources/eventing/README.md (hunk 11 above)
Comment (sayanh) [reply resources/eventing/README.md]: Changed

Comment (sayanh) [reply resources/eventing/README.md]: Changed

**Review by nachtmaar — APPROVED** [review]

File: resources/eventing/README.md [hunk 12]
```diff
@@ -4,25 +4,36 @@ This Helm chart contains all components required for eventing in Kyma:
...
+- controller
 
 ## Event Publisher Proxy
 
-This component receives Cloud Event publishing requests from the cluster workloads (microservice or Serverless functions) and redirects them to the Enterprise Messaging Service Cloud Event Gateway. Click [here](https://github.com/kyma-project/kyma/tree/master/components/event-publisher-proxy) for more details.
+This component receives Cloud Event publishing requests from the cluster workloads (microservice or Serverless functions) and redirects them to the Enterprise Messaging Service Cloud Event Gateway. Click [here](../../components/event-publisher-proxy) for more details.
+
+## Controller
+
+This component manages the internal infrastructure in order to receive an event for all subscriptions
```
Comment (alexandra-simeonova) [inline resources/eventing/README.md:15]:
```suggestion
This component manages the internal infrastructure in order to receive an event for all subscriptions.
```

File: resources/eventing/README.md [hunk 13]
```diff
@@ -4,25 +4,36 @@ This Helm chart contains all components required for eventing in Kyma:
...
 
 ## Install
```
Comment (alexandra-simeonova) [inline resources/eventing/README.md:17]:
```suggestion
## Installation
```

**Review by alexandra-simeonova — DISMISSED** [review]

**Review by nachtmaar — DISMISSED** [review]

**Review by a-thaler — DISMISSED** [review]

**Review by alexandra-simeonova — APPROVED** [review]

**Review by alexandra-simeonova — DISMISSED** [review]

**Review by nachtmaar — DISMISSED** [review]

**Review by a-thaler — APPROVED** [review]

File: resources/cluster-essentials/files/subscriptions.eventing.kyma-project.io.crd.yaml [hunk 14]
```diff
@@ -0,0 +1,186 @@
...
+            properties:
+              filter:
+                description: BebFilters defines the list of BEB filters
+                properties:
+                  dialect:
+                    type: string
+                  filters:
+                    items:
+                      description: BebFilter defines the BEB filter element as a combination
+                        of two CE filter elements
```
Comment (rafalpotempa) [inline resources/cluster-essentials/files/subscriptions.eventing.kyma-project.io.crd.yaml:45]: Do you think this is required?

File: resources/cluster-essentials/files/subscriptions.eventing.kyma-project.io.crd.yaml [hunk 15]
```diff
@@ -0,0 +1,186 @@
...
+                properties:
+                  contentMode:
+                    type: string
+                  exemptHandshake:
+                    type: boolean
+                  qos:
+                    type: string
+                  webhookAuth:
+                    description: Webhook defines the Webhook called by an active subscription
+                      in BEB
```
Comment (rafalpotempa) [inline resources/cluster-essentials/files/subscriptions.eventing.kyma-project.io.crd.yaml:98]: Do you think this is required?

File: resources/cluster-essentials/files/subscriptions.eventing.kyma-project.io.crd.yaml [hunk 16]
```diff
@@ -0,0 +1,186 @@
...
+                - filters
+                type: object
+              id:
+                description: ID is the unique identifier of Subscription, read-only.
+                type: string
+              protocol:
+                type: string
+              protocolsettings:
+                description: ProtocolSettings defines the CE protocol setting specification
+                  implementation
```
Comment (rafalpotempa) [inline resources/cluster-essentials/files/subscriptions.eventing.kyma-project.io.crd.yaml:88]: Do you think this is required?

File: resources/eventing/charts/controller/templates/clusterrole.yaml [hunk 17]
```diff
@@ -0,0 +1,64 @@
...
+  - gateway.kyma-project.io
+  resources:
+  - apirules
+  verbs:
+  - get
+  - list
+  - watch
+  - patch
+  - update
+  - create
```
Comment (rafalpotempa) [inline resources/eventing/charts/controller/templates/clusterrole.yaml:64]: What do you think about introduction of PodSecurityPolicy now? It will be required in 1.19 anyway

File: resources/eventing/Chart.yaml [hunk 18]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v2
 name: eventing
 description: Kyma Component 'eventing'
-version: 1.0.0
```
Comment (janmedrek) [inline resources/eventing/Chart.yaml:4]: I don't get the logic here - why are you "downgrading" the version? Shouldn't it be `apiVersion: v3` and `version: 0.1.0`? Or just `version: 2.0.0`?

File: resources/eventing/charts/controller/values.yaml [hunk 19]
```diff
@@ -0,0 +1,32 @@
...
+podSecurityContext:
+
+bebSecret:
+  nameSuffix: "-beb-oauth2"
+
+metrics:
+  service:
+    nameSuffix: "-metrics"
+
+#TODO(sayanh): Remove this when knative-eventing chart is removed
```
Comment (janmedrek) [inline resources/eventing/charts/controller/values.yaml:31]: Can we get rid of this and just log a GitHub issue or create a follow-up on your board to do this?

File: resources/cluster-essentials/files/subscriptions.eventing.kyma-project.io.crd.yaml (hunk 15 above)
Comment (sayanh) [reply resources/cluster-essentials/files/subscriptions.eventing.kyma-project.io.crd.yaml:98]: It is a valid description of a field. Can you explain your concern in detail?

File: resources/cluster-essentials/files/subscriptions.eventing.kyma-project.io.crd.yaml (hunk 16 above)
Comment (sayanh) [reply resources/cluster-essentials/files/subscriptions.eventing.kyma-project.io.crd.yaml:88]: It is a valid description of a field. Can you explain your concern in detail?

File: resources/eventing/charts/controller/templates/clusterrole.yaml (hunk 17 above)
Comment (sayanh) [reply resources/eventing/charts/controller/templates/clusterrole.yaml:64]: The images are changed to using [gcr.io/distroless/static:nonroot](https://github.com/kyma-project/kyma/blob/master/components/eventing-controller/Dockerfile#L20). Hence PSPs are not relevant here.

File: resources/cluster-essentials/files/subscriptions.eventing.kyma-project.io.crd.yaml (hunk 14 above)
Comment (sayanh) [reply resources/cluster-essentials/files/subscriptions.eventing.kyma-project.io.crd.yaml:45]: It is a valid description of a field. Can you explain your concern in detail?

File: resources/eventing/Chart.yaml (hunk 18 above)
Comment (sayanh) [reply resources/eventing/Chart.yaml:4]: Previously, version 1.0.0 is not correct. Hence we fixed it here with `0.1.0`. This is a chart that is being worked on and it is not installed anywhere right now. ApiVersion seems to be correct as per Helm [docs](https://helm.sh/docs/topics/charts/#the-apiversion-field)
> The apiVersion field should be v2 for Helm charts that require at least Helm 3.

File: resources/eventing/charts/controller/values.yaml (hunk 19 above)
Comment (sayanh) [reply resources/eventing/charts/controller/values.yaml]: Changed

File: resources/cluster-essentials/files/subscriptions.eventing.kyma-project.io.crd.yaml (hunk 16 above)
Comment (rafalpotempa) [reply resources/cluster-essentials/files/subscriptions.eventing.kyma-project.io.crd.yaml:88]: I just think about necessity in having this single words in separate lines.

File: resources/eventing/charts/controller/templates/clusterrole.yaml (hunk 17 above)
Comment (rafalpotempa) [reply resources/eventing/charts/controller/templates/clusterrole.yaml:64]: Great!

File: resources/cluster-essentials/files/subscriptions.eventing.kyma-project.io.crd.yaml (hunk 16 above)
Comment (sayanh) [reply resources/cluster-essentials/files/subscriptions.eventing.kyma-project.io.crd.yaml:88]: The CRDs are generated [here](https://github.com/kyma-project/kyma/blob/master/components/eventing-controller/config/crd/bases/eventing.kyma-project.io_subscriptions.yaml) by kubebuilder which can be manually changed but I prefer not to.

**Review by janmedrek — APPROVED** [review]

**Review by alexandra-simeonova — APPROVED** [review]

**Review by marcobebway — APPROVED** [review]

**Review by Demonsthere — APPROVED** [review]
